### ISSUE TITLE
adding simple music visualizer

### ISSUE THREAD

**Issue by hamidzr** [bruhautomation/ESP-MQTT-JSON-Digital-LEDs#82]
Are people insterested in having [this](https://www.youtube.com/watch?v=5oRir4dck_w) music visualizer added as an option to the list of effects as an option? it would require that you also add a [1$ sound detector](https://www.aliexpress.com/item/Sound-Detector-module/32569653599.html?spm=2114.01010208.3.1.3VT7dF&ws_ab_test=searchweb0_0,searchweb201602_2_10152_10065_10151_10068_10136_10137_10060_10155_10062_437_10154_10056_10055_10054_10059_303_100031_10099_10103_10102_10096_10052_10053_10142_10107_10050_5320017_10051_10171_10084_10083_10080_10082_10081_10110_519_10111_10112_10113_10114_10182_10078_10079_10073_5260016_10123_10189_142,searchweb201603_1,ppcSwitch_3&btsid=83257e9a-0ff4-4a5c-add4-a3fbe65d209e&algo_expid=ae0eb547-0125-4c33-a1ee-9d5605e26b5e-0&algo_pvid=ae0eb547-0125-4c33-a1ee-9d5605e26b5e) to your setup.

**Comment by encor50** [comment]
But how do you add the code ? I dont think it will work as an effect

**Comment by hamidzr** [comment]
i've already merged it with my own use it's possible I can clean it and make a pull request if @bruhautomation thinks this repo is a good place for it.

**Comment by encor50** [comment]
Can you send me your code ? On pastebin or so ? If you want here is my code 
https://github.com/bruhautomation/ESP-MQTT-JSON-Digital-LEDs/issues/32#issuecomment-336731495

**Comment by hamidzr** [comment]
thanks. why didn't you fork and make a PR from it? I will prepare and share it when I get a chance. weekend at the latest.

**Comment by hamidzr** [comment]
I forked the repo and added in my changes. it's available at [my fork](https://github.com/hamidzr/ESP-MQTT-JSON-Digital-LEDs/tree/82-music-visualizer-effect). I'll clean up the code and make a pull request once @bruhautomation approves 

I'm not home so I can't test if I did this correctly. But I am using this at my apartment so it's doable. Let me know if it works for you

**Comment by roandegraaf** [comment]
I implemented it into my code, but when I set it to the visualizer effect in HomeAssistant the ledstrip just turns off and when I turn it on again it's solid. I can't find out where it goes wrong.

**Comment by hamidzr** [comment]
do you have a sound detector attached and working properly?

**Comment by roandegraaf** [comment]
Yes, I tested it on another Arduino and the sound detector does work.

**Comment by hamidzr** [comment]
I assume you also defined the port in the code properly. This is the exact code i'm using in my own setup (minus the passwords, IPs, sensor name and mqtt topics) https://pastebin.com/tXGJb28X take a look and see if it helps. If you managed to find out whats missing from [my fork](https://github.com/hamidzr/ESP-MQTT-JSON-Digital-LEDs/tree/82-music-visualizer-effect) let me know so I can update it.

If you still failed to make it work I encourage you to take a look at the original music visualizer's repo and get that to work first then come back to this.

**Comment by roandegraaf** [comment]
Thank you so much! I'm not home the next two days but when I get home I'll check if I can get it working

**Comment by hamidzr** [comment]
@gfhoihoi72 could you get it to work?

**Comment by fma965** [comment]
I have implemented a music visualizer in my fork. https://github.com/cyanlabs/ESP-MQTT-JSON-Digital-LEDs
